File: RPG City Lite Server/src/main/java/com/jbcomputers/rpgcitylite/dtos/BuildingLevelInfo.java
package com.jbcomputers.rpgcitylite.dtos;

import com.jbcomputers.rpgcitylite.entities.BuildingAction;
import com.jbcomputers.rpgcitylite.entities.BuildingRequirement;
import lombok.Getter;
import lombok.Setter;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.Set;

@Getter
@Setter
public class BuildingLevelInfo {
    Integer buildingId;
    String buildingName;
    String description;
    String image;
    int level;
    Set<BuildingAction> buildingActions;
    Set<BuildingRequirement> buildingRequirements;
}

File: RPG City Lite Client/RPG City Lite/Assets/Scripts/Entities/BuildingLevelInfo.cs
using System.Collections;
using System.Collections.Generic;

public class BuildingLevelInfo
{
    int buildingId;
    string buildingName;
    string description;
    string image;
    int level;
    HashSet<BuildingAction> buildingActions;
    HashSet<BuildingRequirement> buildingRequirements;

    public int BuildingId { get => buildingId; set => buildingId = value; }
    public string BuildingName { get => buildingName; set => buildingName = value; }
    public string Description { get => description; set => description = value; }
    public string Image { get => image; set => image = value; }
    public int Level { get => level; set => level = value; }
    public HashSet<BuildingAction> BuildingActions { get => buildingActions; set => buildingActions = value; }
    public HashSet<BuildingRequirement> BuildingRequirements { get => buildingRequirements; set => buildingRequirements = value; }
}


